### PR TITLE
Merge aspnet/DataProtection into release/2.1 and remove git submodule

### PR DESCRIPTION
**Pull request by natemcmaster**
~~Depends on #1493~~ (Merged)

Part of https://github.com/aspnet/AspNetCore/issues/3598. This merges the history of aspnet/DataProtection repo at its current release/2.1 commit into this repo in the src/DataProtection folder.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -38,10 +38,6 @@
 	path = modules/CORS
 	url = https://github.com/aspnet/CORS.git
 	branch = release/2.1
-[submodule "modules/DataProtection"]
```
Comment (Eilon) [inline .gitmodules]: Yay!!!

**Comment by Eilon** [comment]
Tagging @blowdart - FYI ur codez r moving soon. Mondo repo. It's happening.

**Comment by natemcmaster** [comment]
> Please wait until after tomorrow's shiproom

This only touches the 2.1 branch. Is it still okay to merge? I'll hold off on bringing this up into 2.2 until after shiproom. The 2.2 merge is in https://github.com/aspnet/Universe/pull/1497

**Comment by Eilon** [comment]
Ah OK then that's fine as long as it has ZERO impact on 2.2.

**Comment by natemcmaster** [comment]
Correct, ZERO impact on 2.2. Nothing automatically merges back into 2.2, so this only affects 2.1 builds.

**Pull request merged**